sinon/lib/util/Wrapper.py

```diff
@@ -89,53 +88,136 @@ def wrapped(*args, **kwargs):
     wrapped.LOCK = True
     return wrapped
 
-def __gen_index_list(condition, *args, **kwargs):
-    # Generating index list based on arguments and condition
-    # Note: ignore args[0] because it is callback in this condition
-    # combination
+def __gen_index_list(conditions, *args, **kwargs):
+    """
+    Args:
+        conditions: dictionary, the SinonStub's conditions (the user-defined behaviour for the stub)
+        args: tuple, the arguments inputed by the user
+        kwargs: tuple, the keyword arguments inputed by the user
+    Returns:
+        list, the list of indices in conditions for which the user args/kwargs match
+    """
+    return __get_call_indices(conditions["target"], args, kwargs, conditions["args"], conditions["kwargs"])
 
+def __get_call_indices(target, args, kwargs, args_list, kwargs_list):
```

Comment:
You can pass 3 parameters: args, kwargs, conditions
And use conditions['target], conditions['args'], conditions['kwargs'] for making code cleaner

Reply:
The values for `args_list` and `kwargs_list` can come from the `wrapper` or from the `conditions`. That's why `args_list` and `kwargs_list` are passed in explicitly, to support both cases.